lib/Doctrine/Annotations/Metadata/PropertyMetadata.php

```diff
@@ -4,38 +4,52 @@
 
 namespace Doctrine\Annotations\Metadata;
 
-use Doctrine\Annotations\Metadata\Type\Type;
+use Doctrine\Annotations\Metadata\Constraint\Constraint;
+use Doctrine\Annotations\Metadata\Constraint\ConstraintNotFulfilled;
 
 final class PropertyMetadata
 {
     /** @var string */
     private $name;
 
-    /** @var Type */
-    private $type;
```

Comment:
As discussed, Constraint should not replace `$type` here but build on top of it instead.